### PR TITLE
add marvin plugin

### PR DESCRIPTION
**Pull request by matheusfm**
Add [marvin](https://github.com/undistro/marvin) plugin to index

### CODE REVIEW

**Comment by ahmetb** [comment]
Thanks for the submission. I see this tool was mostly designed as a standalone tool today. What's your motive distributing it as a kubectl plugin? Typically kubectl plugins enhance existing, add missing functionality to kubectl, and when users type --help, they expect to see the plugin usage as `kubectl marvin`. Is this just another distribution method for you?

**Comment by matheusfm** [comment]
Hi @ahmetb, thank you for the revision.

>  What's your motive distributing it as a kubectl plugin? Typically kubectl plugins enhance existing, add missing functionality to kubectl

Using marvin as a kubectl plugin, the scanning functionality would be added to kubectl. 
The users would be able to scan the k8s cluster by running `kubectl marvin scan`.

It's similar to [popeye](https://github.com/derailed/popeye) (which also it's a kubectl plugin). Marvin is an alternative, where users can easily write [**custom checks**](https://github.com/undistro/marvin#custom-checks) with [CEL](https://github.com/google/cel-spec).
This way of checking k8s resources could become popular because the new [k8s `ValidatingAdmissionPolicy` API](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/) also uses CEL. Marvin checks are heavily inspired in this API.

> and when users type --help, they expect to see the plugin usage as `kubectl marvin`.

I have updated the usage messages to add `kubectl` prefix. See my tests:
```shell
matheusfm:~/Workspace/krew-index (plugin/marvin)$ kubectl krew install --manifest plugins/marvin.yaml --archive /tmp/marvin_Linux_x86_64.tar.gz 
Installing plugin: marvin
Installed plugin: marvin
\
 | Use this plugin:
 | 	kubectl marvin
 | Documentation:
 | 	https://github.com/undistro/marvin
/
matheusfm:~/Workspace/krew-index (plugin/marvin)$ kubectl marvin --help
A Kubernetes cluster scanner

Usage:
  kubectl marvin [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  scan        Scan a Kubernetes cluster
  version     Show the version of Marvin

Flags:
  -h, --help       help for marvin
      --no-color   Disable color output
  -v, --v Level    number for the log level verbosity

Use "kubectl marvin [command] --help" for more information about a command.
matheusfm:~/Workspace/krew-index (plugin/marvin)$ kubectl marvin scan --help
Scan a Kubernetes cluster

Usage:
  kubectl marvin scan [flags]

Examples:
  # Scan the current cluster
  kubectl marvin scan

  # Scan the 'foo' namespace of the current cluster
  kubectl marvin scan -n foo

  # Scan the current cluster providing custom checks
  kubectl marvin scan --checks ./examples/

  # Scan the current cluster providing custom checks and disabling the built-in checks
  kubectl marvin scan --disable-builtin --checks ./examples/

  # Scan a specific cluster using a kubeconfig file
  kubectl marvin scan --kubeconfig /path/to/kubeconfig.yml

  # Scan the current cluster, but do not fail even if there are errors in the report
  kubectl marvin scan --no-fail

  # Scan the current cluster and generate output in JSON format
  kubectl marvin scan -o json

Flags:
      --as string                      Username to impersonate for the operation. User could be a regular user or a service account in a namespace.
      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --as-uid string                  UID to impersonate for the operation.
      --cache-dir string               Default cache directory (default "/home/matheusfm/.kube/cache")
      --certificate-authority string   Path to a cert file for the certificate authority
  -f, --checks string                  Path to the check files directory
      --client-certificate string      Path to a client certificate file for TLS
      --client-key string              Path to a client key file for TLS
      --cluster string                 The name of the kubeconfig cluster to use
      --context string                 The name of the kubeconfig context to use
      --disable-annotation-skip        Disable resource skipping by annotation
      --disable-builtin                Disable builtin checks
      --disable-compression            If true, opt-out of response compression for all requests to the server
  -h, --help                           help for scan
      --insecure-skip-tls-verify       If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
      --kubeconfig string              Path to the kubeconfig file to use for CLI requests.
  -n, --namespace string               If present, the namespace scope for this CLI request
      --no-fail                        Return an exit code of zero even if there are errors in the report
  -o, --output string                  Output format. One of: ("table", "json", "yaml" or "markdown") (default "table")
      --request-timeout string         The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
  -s, --server string                  The address and port of the Kubernetes API server
      --skip-annotation string         Annotation name for skipping checks (default "marvin.undistro.io/skip")
      --tls-server-name string         Server name to use for server certificate validation. If it is not provided, the hostname used to contact the server is used
      --token string                   Bearer token for authentication to the API server
      --user string                    The name of the kubeconfig user to use

Global Flags:
      --no-color   Disable color output
  -v, --v Level    number for the log level verbosity
```

File: plugins/marvin.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: marvin
+spec:
+  homepage: https://github.com/undistro/marvin
+  shortDescription: Scans your cluster for potential issues. Supports custom checks with CEL.
```
Comment (ahmetb) [inline plugins/marvin.yaml:7]: Shorten to 50 chars as it gets truncated otherwise.

File: plugins/marvin.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  description: |
+    Marvin scans a k8s cluster by performing CEL expressions
```
Comment (ahmetb) [inline plugins/marvin.yaml:9]: You can spell out Kubernetes, no need to abbreviate.

**Review by ahmetb — COMMENTED** [review]
Thanks for the explanation.

File: plugins/marvin.yaml (hunk 1 above)
Comment (matheusfm) [reply plugins/marvin.yaml]: Ok, I've shorten it. Thanks.

File: plugins/marvin.yaml (hunk 2 above)
Comment (matheusfm) [reply plugins/marvin.yaml]: Done

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**